backend-project/small_eod/notifications/views.py

```diff
@@ -0,0 +1,68 @@
+from django.forms.models import model_to_dict
+from rest_framework.views import APIView
+
+
+class SendNotificationsMixin(APIView):
+    notified_users = ""
+    initial_instance = None
+
+    def send_notifications(self, http_action, **kwargs):
+        instance = None
+
+        if http_action in ["post", "put"]:
+            self.kwargs["pk"] = kwargs["data"]["id"]
+            instance = self.get_object()
+        else:
+            instance = self.initial_instance
+
+        notified_users = self.get_user_list(instance)
+        sender = self.basename
+
+        for user in notified_users:
+            user.notify(sender, self.action, **kwargs)
+
+    def get_user_list(self, instance):
```

Comment:
Detal, ale w przypadku braku nadpisania `notified_users` to wygeneruje mało czytelny komunikat błędu. Przed wcieleniem warto zacząć wyjątek ze wskazaniem co musi być nadpisane. W wielu mixinach to już mamy.